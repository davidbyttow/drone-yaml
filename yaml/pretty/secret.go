package pretty

import (
	"sort"

	"github.com/drone/drone-yaml/yaml"
)

// TODO consider "!!binary |" for secret value

// helper function to pretty prints the signature resource.
func printSecret(w writer, v *yaml.Secret) {
	w.WriteString("---")
	w.WriteTagValue("kind", v.Kind)
	w.WriteTagValue("type", v.Type)
	if v.Type == "encrypted" {
		printData(w, v.Data)
	} else {
		w.WriteTagValue("data", v.Data)
	}
	w.WriteByte('\n')
	w.WriteByte('\n')
}

func printData(w writer, d map[string]string) {
	var keys []string
	for k := range d {
		keys = append(keys, k)
	}
	sort.Strings(keys)

	w.WriteTag("data")
	w.IndentIncrease()
	for _, k := range keys {
		v := d[k]
		w.WriteTag(k)
		w.WriteByte(' ')
		w.WriteByte('>')
		w.IndentIncrease()
		for _, s := range chunk(v, 60) {
			w.WriteByte('\n')
			w.Indent()
			w.WriteString(s)
		}
		w.IndentDecrease()
	}
	w.IndentDecrease()
}
