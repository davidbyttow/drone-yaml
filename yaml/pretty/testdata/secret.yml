kind: secret
type: encrypted

data:
  username: N2NmYjA3ODQwNTY1ODFlY2E5MGJmOWI1NDk0NDFhMTEK
  password: NGZhNjY5YWMxZjhlYzJkNzE1ODlkZDliN2I4MDMwOTEzNGZhZTk3ZjcyNzk5NzNmZmQ3ZWRmNGY0YWJmYjFlMGY3ZmI2MmQ2MmNjMDQ1NDQwNmU5Nzc5NTlmNDEyYzM2YzI1ZjdhOWVkOTc1OTI5YmE5OTY1ZGRhOTk3NTQ1NDAK
