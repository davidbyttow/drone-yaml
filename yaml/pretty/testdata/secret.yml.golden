---
kind: secret
type: general
data:
  password: >
    NGZhNjY5YWMxZjhlYzJkNzE1ODlkZDliN2I4MDMwOTEzNGZhZTk3ZjcyNzk5
    NzNmZmQ3ZWRmNGY0YWJmYjFlMGY3ZmI2MmQ2MmNjMDQ1NDQwNmU5Nzc5NTlm
    NDEyYzM2YzI1ZjdhOWVkOTc1OTI5YmE5OTY1ZGRhOTk3NTQ1NDAK
  username: >
    N2NmYjA3ODQwNTY1ODFlY2E5MGJmOWI1NDk0NDFhMTEK

...
